src/librustc/hir/lowering.rs

```diff
@@ -321,6 +321,47 @@ enum AnonymousLifetimeMode {
 
     /// Pass responsibility to `resolve_lifetime` code for all cases.
     PassThrough,
+
+    /// Used in the return types of `async fn` where there exists
+    /// exactly one argument-position elided lifetime.
+    ///
+    /// In `async fn`, we lower the arguments types using the `CreateParameter`
+    /// mode, meaning that non-`dyn` elided lifetimes are assigned a fresh name.
+    /// If any corresponding elided lifetimes appear in the output, we need to
```

Comment:
Is this paragraph truncated, or does it lead into the next comments? If the latter, that could be made clearer (e.g. ending this sentence with a colon and putting the next two paragraphs in bullet points).